src/modules/mount/main.py

```diff
@@ -274,6 +274,11 @@ def mount_partition(root_mount_point, partition, partitions, mount_options, moun
 
         # Store created list in global storage so it can be used in the fstab module
         libcalamares.globalstorage.insert("btrfsSubvolumes", btrfs_subvolumes)
+
+        # Don't do any subvolume mounting or unmounting if we don't have any
+        if btrfs_subvolumes == ["mountPoint": "/", "subvolumes": ""]:
+            break
+
```

Comment:
This code isn't needed.  It is not a general solution.

The code you removed was a general solution I think.

Reply:
with this and not the top level dictionary check it will unmount / on line 292. We need a way to avoid that. I'll put this back in, but we need some way to not unmount the / mountPoint when not using subvolumes.

Reply:
Ok, added back in the subvolume mounting continue, but something also needs to skip the subprocess umount for my use case.

Reply:
Yes, don't do the umount if there is a subvolume `""`

Reply:
sorry, rereading the changes I made and the overal issue. The problem is "just" the "subvol=," argument that gets added when we mount the volume lower here.  There's a umount call above it, which made me think by this point / was already mounted, but needed to be unmounted only for the cases where there's a subvolume so that can get mounted properly. But looking at the other fstype cases we probably need to mount it in here, just without the bad "subvol=," argument. I'll rework my change to try to address that instead.

Reply:
Ok, I think what I pushed should fix the problem I hit. I'll test on a usb stick now.

Reply:
> Yes, don't do the umount if there is a subvolume `""`

Well, when I had that in place you suggested it wasn't needed. Really it's not, as long as we don't try to remount with "subvol=," it should work to unmount then remount. Testing on an image with my latest changes now. but I think it covers the 1 issue I hit last time and the other issue I hit when trying a fix.

Reply:
Yep, code as is is working. It does unmount but then it remounts without the subvol=, argument so it does remount and does have the filesystem to sync into.